### ISSUE TITLE
Mentions in WallPost not working correctly

### ISSUE THREAD

**Issue by queermeet-dev** [opensource-socialnetwork/opensource-socialnetwork#2391]
I just made friends with someone on my website, when I try to mention that person in a a wall post, the person is not shown, even though we are friends.

I already logged out and in again, but no success.

Tried on PC and on mobile -> other persons are shown

Is that a bug?

**Comment by lianglee** [comment]
How did you mentioned? Using a mentions component?

**Comment by queermeet-dev** [comment]
No there is this symbol in the wall container
Am 5. Aug. 2024, 11:08 +0200 schrieb Arsalan Shah ***@***.***>:

**Comment by lianglee** [comment]
Which symbol share screenshot please?

**Comment by queermeet-dev** [comment]
![image](https://github.com/user-attachments/assets/8f0bbc34-3905-4b67-b91a-84403523175b)

**Comment by githubertus** [comment]
trying to understand your report ... you wrote "Tried on PC and on mobile -> other persons are shown"

that is: 
you click that friends icon
the input field opens
you enter the first character of your friend's name
but that friend does NOT appear in the list, but any other members DO appear ?!?

**Comment by queermeet-dev** [comment]
Yes, when I type "L" another friend appears, but not the new friend
Am 6. Aug. 2024, 12:50 +0200 schrieb Michael Zülsdorff ***@***.***>:

**Comment by queermeet-dev** [comment]
I've 3 friends with "l"

This is the search

![image](https://github.com/user-attachments/assets/3a6d3d09-d3c9-4053-a828-520fd84a64b8)


This is the other 2 friends 


![image](https://github.com/user-attachments/assets/f6485a4a-531f-45c6-8d80-70696434f9a2)

**Comment by queermeet-dev** [comment]
hey, any info on this???

**Comment by lianglee** [comment]
Delete friend and try to add again.
Also try to disable display username and see if it works?

**Comment by queermeet-dev** [comment]
I tried disable displayusername and it seems to work and after re-enabling it does not work anymore

I tried with 2 users, with displayusername they are not shown, without displayusername they are shown

**Comment by githubertus** [comment]
Sorry, but I'm busy with other stuff until the end of september, so just as quick hint:
Arsalan applied a change to the friendspicker of OssnWall some time ago that didn't make it into DisplayUserName
Most likely it needs an adapted extra option ...

**Comment by queermeet-dev** [comment]
okay, thanks.

**Comment by lianglee** [comment]
What was the issue? How you fixed it?

**Comment by queermeet-dev** [comment]
I gave the changed Code from OSSNWall and the DisPlayUsername code to ChatGPT andI tried to fix it but I could not work it out , so I changed the function com_display_username_friend_picker() in displayusername, so you can now mention every user in a post, not jsut friends


that was the original one


```
function com_display_username_friend_picker()
{
	header('Content-Type: application/json');
	if (!ossn_isLoggedin()) {
		exit;
	}
	$user    = new OssnUser;
	$friends = $user->getFriends(ossn_loggedin_user()->guid);
	if (!$friends) {
		return false;
	}
	$search_for = input('q');
	// allow case insensitivity with first typed in char
	$fc = mb_strtoupper(mb_substr($search_for, 0, 1,'UTF-8'), 'UTF-8');
	$search_For = $fc . mb_substr($search_for, 1, null, 'UTF-8');
	// show all friends with wildcard '*' in first place
	if ($search_for == '*') {
		$search_for = '';
		$search_For = '';
	}
	$search_len = mb_strlen($search_for,'UTF-8');
	$usera = array();
	foreach ($friends as $users) {
		$first_name_start = mb_substr($users->username, 0, $search_len, 'UTF-8');
		if ($first_name_start == $search_for || $first_name_start == $search_For) {
			$p['first_name'] = $users->username;
			$p['last_name']  = '';
			$p['imageurl']   = ossn_site_url("avatar/{$users->username}/smaller");
			$p['id']         = $users->guid;
			$usera[]         = $p;
		}
	}
	echo json_encode($usera);
}

```


and that is ChatPGT Change


```
function com_display_username_friend_picker()
{
 header('Content-Type: application/json');
 if (!ossn_isLoggedin()) {
 exit;
 }

 $search_for = trim(input('q')); // Trim to remove any accidental spaces

 // Convert to lowercase for case-insensitive comparison
 $search_for = mb_strtolower($search_for, 'UTF-8');

 $user = new OssnUser;

 // Fetch all users instead of just friends
 $users = $user->searchUsers(array(
 'wheres' => "LOWER(u.username) LIKE '%{$search_for}%'",
 ));

 $usera = array();
 if ($users) {
 foreach ($users as $u) {
 $p['first_name'] = $u->username;
 $p['last_name'] = ''; // Must be present, even if empty
 $p['imageurl'] = ossn_site_url("avatar/{$u->username}/smaller");
 $p['id'] = $u->guid;
 $usera[] = $p;
 }
 }

 echo json_encode($usera);
}
```

**Comment by githubertus** [comment]
The issue should be fixed with DisplayUsername 7.4.
thx for reporting
